modules/annotations/src/main/java/org/mule/config/processors/DecoratingAnnotatedServiceProcessor.java

```diff
@@ -79,20 +82,19 @@ public DecoratingAnnotatedServiceProcessor(MuleContext context)
 
     public void setMuleContext(MuleContext context)
     {
-        try
-        {
-            this.context = context;
-            this.regProps = new RegistryMap(context.getRegistry());
-            this.helper = new AnnotatedEndpointHelper(context);
-            this.parserFactory = context.getRegistry().lookupObject(AnnotationsParserFactory.class);
-            if(parserFactory==null)
-            {
-                logger.info(AnnotationsParserFactory.class.getName() +" implementation not found in registry, annotations not enabled");
-            }
-        }
-        catch (MuleException e)
+        this.context = context;
+        this.regProps = new RegistryMap(context.getRegistry());
+    }
+
+
+    @Override
+    public void start() throws MuleException
+    {
+        helper = new AnnotatedEndpointHelper(context);
+        parserFactory = context.getRegistry().lookupObject(AnnotationsParserFactory.class);
+        if (parserFactory == null)
         {
-            throw new MuleRuntimeException(CoreMessages.failedToCreate(getClass().getName()), e);
+            logger.info(AnnotationsParserFactory.class.getName() + " implementation not found in registry, annotations not enabled");
```

Comment:
Because it's valid that the AnnotationsParserFactory is not present. In the past, it didn't fail because when it wasn't present, neither was the post processor. Also, the function that the parser factory did is now provided by Spring OOTB.